### PR TITLE
apiversionalist

### PR DESCRIPTION
**Pull request by wolferen**

### CODE REVIEW

File: app/controllers/api/v1/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      end
+
+      def show
+        product = Product.friendly.find(params[:id])
+        render json: { status: 'SUCCESS', message: "Load #{product.slug} product", data: product }, status: :ok
+      end
+
+      def create
+        @product = Product.new(product_params)
+        @product.category_id = params[:category_id]
```
Comment (yuriy) [inline app/controllers/api/v1/products_controller.rb]: @wolferen why are you doing this? What about permition in `product_params`?

File: app/controllers/api/v1/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        render json: { status: 'SUCCESS', message: 'Load all produts', data: products }, status: :ok
+      end
+
+      def show
+        product = Product.friendly.find(params[:id])
+        render json: { status: 'SUCCESS', message: "Load #{product.slug} product", data: product }, status: :ok
+      end
+
+      def create
+        @product = Product.new(product_params)
```
Comment (yuriy) [inline app/controllers/api/v1/products_controller.rb]: @wolferen why do you use here instance variable but in the `index`, `show` methods you don't?

File: spec/requests/v1/base_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+describe V1::BaseController do
```
Comment (yuriy) [inline spec/requests/v1/base_controller_spec.rb]: @wolferen do you have `BaseController` ?

File: spec/controllers/v1/products_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+describe V1::ProductsController do
```
Comment (yuriy) [inline spec/controllers/v1/products_controller_spec.rb]: @wolferen what about specs? Can you add few tests to check the versionist.

File: spec/requests/v1/products_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Api
+  module V1
+    describe ProductsController do
+      it 'should get v1' do
+        get '/api/v1/products'
```
Comment (yuriy) [inline spec/requests/v1/products_controller_spec.rb]: @wolferen stop! we've discussed that api version should be in headers and not in endpoint path.

Comment (wolferen) [reply spec/requests/v1/products_controller_spec.rb]: @yuriy ah, forgot about this. Will do.

File: spec/requests/v1/products_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+require 'rails_helper'
+
+module Api
+  module V1
+    describe ProductsController do
+      it 'should get api v1' do
+        get '/products/'
+        expect(response).to be_success
+      end
```
Comment (yuriy) [inline spec/requests/v1/products_controller_spec.rb]: @wolferen add please also tests for cases when request goes without version headers and when the version is wrong

**Comment by yuriy** [comment]
@wolferen also `squash` a bunch of commits into one. `1 PR -> 1 Commit`.
This is a good practice, b/c when you'll need to rebase your PR (with frew commits) with master branch, it could take more time and can cause some conflicts.

**Comment by yuriy** [comment]
@wolferen after `squash` ready to go

**Pull request merged**